pkg/store/telemetry.go

```diff
@@ -67,18 +67,24 @@ func (s *seriesStatsAggregator) reset() {
 	s.seriesStats = storepb.SeriesStatsCounter{}
 }
 
-func (s *seriesStatsAggregator) findBucket(value float64, quantiles []int64) int64 {
+func (s *seriesStatsAggregator) findBucket(value float64, quantiles []int64) string {
 	if len(quantiles) == 0 {
-		return 0
+		return "+Inf"
 	}
-	var foundBucket int64
+
+	var foundBucket int64 = -1
 	for _, bucket := range quantiles {
-		foundBucket = bucket
 		if value < float64(bucket) {
+			foundBucket = bucket
 			break
 		}
 	}
-	return foundBucket
+
+	if foundBucket == -1 {
+		return "+Inf"
+	} else {
+		return strconv.Itoa(int(foundBucket))
+	}
 }
```

Comment:
Regarding the type things, it's a little messy, I agree.

The command line args for serier/buckets take a `[]int64`
[However the type which count the series/sample size of queries are `int`](https://github.com/thanos-io/thanos/blob/cdb395a7100be554e804d61c735b8d4a4b678f11/pkg/store/storepb/custom.go#L484)
To compare them we then do `int` -> `float64` and `int64` to `float64` 

I'm not sure why we use floats in the first place here, these values will never be decimals.

Maybe we could change the `int` in the storepb to `int64` but seems like that would be a larger effort.

Kingpin doesn't have "un-sized" ints (i.e one can do int16. int32, int64).

Why don't we just do a `int` -> `int64` cast, and avoid casting to float?

Then we just need to following cast:

```
seriesLeBucket := s.findBucket(int64(s.seriesStats.Series), s.seriesLeBuckets)
samplesLeBucket := s.findBucket(int64(s.seriesStats.Samples), s.samplesLeBuckets)
...
func findBucket(value int64, quantiles []int64) string {
```

Reply:
Ack! Let me clarify! The reason why I recommend using float64 for the metric buckets is to follow similar conventions to client_golang, where `(prometheus.HistogramOpts).Buckets` is `[]float64`. This is already a precedent i.e, a 2D histogram, so it'd be great to avoid further breaking conventions.

What I suggested is just converting the `seriesLeBuckets` and `samplesLeBuckets` to `[]float64` instead of `[]int64`. We won't need to touch the aggregation login in `storepb` this way, and only need to make the kingpin flags `Float64List()` and convert series stats from int -> float64 in `Observe` as we did earlier. 🙂

Reply:
Alright fair enough, although I find it a bit strange i use floats here.

Fixed up the commits.